### PR TITLE
Camouflage: Deadborder Pack, Base Component Swapover

### LINKED ISSUE #2650: CamoChooser problems
Unless this is already corrected on the master:

* The CamoChooser does not react to double clicking anymore.
* Selecting another directory, then NOT selecting a camo and clicking OK throws an NPE.

### PR DESCRIPTION
**Pull request by Windchild292**
This swaps our base camouflage pack from SeaVee's to Deadborder's, and migrates the dialogs to base components. Further, it adds a way to read/write to XML, which is used in MHQ.

For review:
The code is entirely contained in the commits before I added the camouflage pack.
The goal is for this to be merged in early in 0.49.3's dev cycle.

This fixes #2650 during the swapover, and should properly work for use with force icons.

### CODE REVIEW

File: megamek/src/megamek/client/ui/swing/dialog/imageChooser/AbstractIconChooser.java [hunk 1]
```diff
@@ -85,8 +85,6 @@ private void initialize(JTree tree) {
         splitPane = new JSplitPane(JSplitPane.HORIZONTAL_SPLIT, true, scrpTree, scrpImages);
         splitPane.setResizeWeight(0.5);
 
-        splitPane.setDividerLocation(GUIPreferences.getInstance().getInt(GUIPreferences.IMAGE_CHOOSER_SPLIT_POS));
```
Comment (NickAragua) [inline megamek/src/megamek/client/ui/swing/dialog/imageChooser/AbstractIconChooser.java:88]: ?

**Review by NickAragua — APPROVED** [review]
One question in the code. I have no meaningful way of reviewing the data. And also my browser locks up when trying to look at the code review in full.

File: megamek/src/megamek/client/ui/swing/dialog/imageChooser/AbstractIconChooser.java (hunk 1 above)
Comment (Windchild292) [reply megamek/src/megamek/client/ui/swing/dialog/imageChooser/AbstractIconChooser.java:88]: Missed the MMPreferences side of the swapover, nice catch.

**Comment by Windchild292** [comment]
Great catch. My browser does the same for a bit, but then loads. I'd assume it is likely due to the number of data file changes. As for the data review... I've done a spot check of the new camouflage pack and it looked good to me.

**Comment by Windchild292** [comment]
Forgot to mention originally... this requires https://github.com/MegaMek/mekhq/pull/2622 to merge, as otherwise MHQ won't build.

**Pull request merged**